org.jacoco.core/src/org/jacoco/core/internal/analysis/MethodAnalyzer.java

```diff
@@ -86,6 +99,38 @@ public IMethodCoverage getCoverage() {
 		return coverage;
 	}
 
+	/**
+	 * {@link MethodNode#accept(MethodVisitor)}
+	 */
+	@Override
+	public void accept(final MethodNode methodNode,
+			final MethodVisitor methodVisitor) {
+		this.ignored.clear();
+		for (final IFilter filter : FILTERS) {
+			filter.filter(methodNode, this);
+		}
+
+		for (final TryCatchBlockNode n : methodNode.tryCatchBlocks) {
+			n.accept(methodVisitor);
+		}
+		for (int i = 0; i < methodNode.instructions.size(); i++) {
```

Comment:
And one more suggestion 😆  - `InsnList` as well as whole `MethodNode` comes to us via `MethodSanitizer#visitEnd()` in `ClassProbesAdapter`, where we can replace value of field `instructions` (that is public in `MethodNode`) by a new instance with overriden method `toArray()`.

> But I woudn't worry too much right now - unless performance testing shows a bottle neck here.

I've checked that our code doesn't trigger creation of array anymore.
So indeed - let's don't worry about this right now.